src/main.c

```diff
@@ -24,71 +24,74 @@
 
 static void start_scan(void);
 
+K_WORK_DELAYABLE_DEFINE(scan_work, start_scan);
+
 static struct bt_conn *conn_connecting = NULL;
-static uint8_t volatile conn_count;
-static bool volatile is_connecting = false;
+static uint8_t volatile conn_count = 0;
 
 static bool find_device_name(struct bt_data *data, void *user_data)
 {
-    char* device_name = user_data;
+    bt_addr_le_t *addr = user_data;
+    char device_name[31];
 
     // a return value of "false" will stop data parsing by bt_data_parse
 
     if (data->type == BT_DATA_NAME_COMPLETE) {
         memcpy(device_name, data->data, data->data_len);
         device_name[data->data_len] = '\0';
-        return false;
-    }
-    else {
-        return true;
-    }
+        
+        if (strcmp(device_name, "DXC") == 0) {
+
+            int err;
+
+            err = bt_le_scan_stop();
+            if (err) {
+                printk("Stop LE scan failed (err %d)\n", err);
+                // I am unsure the of the proper way to handle a
+                // bt_le_scan_stop failure but I know a connection should not be 
+                // attempted if it fails
+                continue;
```

Comment:
@jakejunk150  No worried about it. You can ignore it from now. In general, it should be a bug or work-around to force BLE stop then re-init.